src/components/BestProducts.jsx

```diff
@@ -0,0 +1,63 @@
+import { getProducts } from "../api/productApi";
+import { useState, useEffect } from "react";
+import "./BestProducts.css";
+
+/**
+ *
+ * 베스트 상품을 보여주는 컴포넌트
+ * 인기순(favorite)으로 정렬된 상품을 보여줍니다. (반응형 크기에 따라 PC: 4 / tablet: 2 / mobile: 1)
+ * resize 함수를 사용하여 화면 크기에 따라 visibleCount를 조절하여  상품을 보여줍니다.
+ */
+const BestProduct = () => {
+  const [bestProducts, setBestProducts] = useState([]);
+  const [visibleCount, setVisibleCount] = useState(4); // default pc
+
+  useEffect(() => {
+    getProducts("?page=1&pageSize=4&orderBy=favorite").then((data) => {
+      setBestProducts(data.list);
+    });
+  }, []);
+
+  useEffect(() => {
+    const updateVisibleCount = () => {
+      const width = window.innerWidth;
+
+      if (width <= 767) {
+        // mobile
+        setVisibleCount(1);
+      } else if (width <= 1199) {
+        // tablet
+        setVisibleCount(2);
+      } else {
+        setVisibleCount(4);
+      }
+    };
+
+    updateVisibleCount();
+    window.addEventListener("resize", updateVisibleCount); // resize 시 eventListener
+
+    return () => window.removeEventListener("resize", updateVisibleCount);
+  }, []);
```

Comment:
💬 여담
useEffect 내부에 선언된 updateVisibleCount 함수는 컴포넌트 바깥으로 분리하셔도 됩니다.
외부로 updateVisibleCount 함수를 분리하게 되면 useEffect 의존배열에 updateVisibleCount가 들어가게 되니
해당 함수가 재정의되지 않도록 useCallback으로 감싸셔야합니다~

```suggestion
  const updateVisibleCount = useCallback(() => {
    const width = window.innerWidth;

    if (width <= 767) {
      setVisibleCount(1);
    } else if (width <= 1199) {
      setVisibleCount(2);
    } else {
      setVisibleCount(4);
    }
  }, []);
    
  useEffect(() => {
    updateVisibleCount();
    window.addEventListener("resize", updateVisibleCount);

    return () => window.removeEventListener("resize", updateVisibleCount);
  }, [updateVisibleCount]);
```

Reply:
아하 제가 useCallback을 잘몰라서 좀 더 사용예제를 찾아보고 적용시키도록 하겠습니다!!! 감사합니다 ㅎㅎㅎ